src/military/military.cpp

```diff
@@ -2900,6 +2900,11 @@ void implement_war_goal(sys::state& state, dcon::war_id war, dcon::cb_type_id wa
 	float prestige_gain = successful_cb_prestige(state, wargoal, from) * (war ? 1.0f : state.defines.crisis_wargoal_prestige_mult);
 	nations::adjust_prestige(state, from, prestige_gain);
 	nations::adjust_prestige(state, target, -prestige_gain);
+
+	// If the nation stops existing we shall cleanup all wars it was involved at
+	if(state.world.nation_get_owned_province_count(target) == 0)
```

Comment:
This check shouldn't be here; it should be done when a nation loses a province (provinces can also be lost via event or rebel defection, for example)